### PR TITLE
feat: add construct method to retrieve generated function slots

### PR DESCRIPTION
**Pull request by alharris-at**
#### Description of changes
Add construct method to retrieve generated function slots. This should make it more feasible to construct overrides, by first inspecting the generated results, then providing those objects back as overrides.

##### CDK / CloudFormation Parameters Changed
N/A

#### Issue #, if available
N/A

#### Description of how you validated changes
Unit tests

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by AaronZyLee — APPROVED** [review]

**Review by phani-srikar — APPROVED** [review]

File: packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts [hunk 1]
```diff
@@ -113,4 +122,24 @@ export class AmplifyGraphqlApi extends Construct {
...
     this.resources = generateConstructExports(transformedResources.rootStack, transformedResources.stacks, transformerStack);
   }
+
+  /**
+   * Get the function slots generated by the GraphQL transform operation, adhering to the FunctionSlot interface.
+   * @returns the list of generated function slots in the transformer, in order to facilitate overrides.
+   */
+  getGeneratedFunctionSlots(): FunctionSlot[] {
+    return Object.entries(this.transformerGeneratedResolvers)
+      .filter(([name]) => name.split('.').length === 6)
```
Comment (phani-srikar) [inline packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts:132]: Some field level resolvers like `Model.Field.vtl` do not confirm to this format. Are we leaving them out here?

**Review by phani-srikar — COMMENTED** [review]
Some field level resolvers like Model.Field.vtl do not confirm to this format. Are we leaving them out here?

File: packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts (hunk 1 above)
Comment (phani-srikar) [inline packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts:132]: I meant here sorry.

Comment (alharris-at) [reply packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts:132]: Yeah, since they can't be overridden via the slot system today.

**Pull request merged**